source/blog/2017-02-16-hyperconverged-ovirt-with-ovn-part2.md

```diff
@@ -0,0 +1,164 @@
+---
+title: HyperConverged oVirt (GlusterFS) with OVN (OpenVSwitch + OpenFlow + Tunnels) provider (part 2)
+author: stirabos
+tags: community, documentation, howto, glusterfs, gluster, hyperconverged, hyper-converged, hosted-engine, OVN, SDN, demo, cockpit
+date: 2017-02-16 21:00:00 CET
+comments: true
+published: false
+---
+
+How-to build a demo env for HyperConverged oVirt with OVN (part 2).
+
+In part 1 we deployed an hyper-converged GlusterFS based ovirt cluster, now we'll see how to add the OVN provider to this setup.
+
+## oVirt OVN Provider
+
+OVN - Open Virtual Network - is an OVS (Open vSwitch) extension, adding support for virtual networks abstraction. It adds native OVS support for virtual L2 and L3 overlays. The oVirt OVN Provider allow using OVN provided networks from within oVirt, using the external network provider mechanism. We will be able to use/define OVN logical networks from within oVirt, and provision virtual machines with network interfaces connected to the OVN networks.
+
+The oVirt OVN Provider is composed by *ovirt-provider-ovn* to be installed on a central system (the engine VM is our best option for our simple scenario) and *ovirt-provider-ovn-driver* to be installed on each of the involved hosts.
+The ovirt-provider-ovn requires openvswitch-ovn-central which provides the OVN Northbound DB (contains the high level logical network configuration) - and the OVN Southbound DB (contains the physical network view) plus the required daemons to translate data and requests.
+The ovirt-provider-ovn and openvswitch-ovn-central are required to define and edit logical network configuration but they are not required for the routing process which is instead distributed in the OVN world so they are not a single points of failure.
+
+oVirt cannot yet directly install and configure ovirt-provider-ovn and ovirt-provider-ovn-driver so manual actions are required today to install and configure them; for simplicity we are now going to disable the firewall on the engine VM and on the hosts. A proper install mechanism and automated handling of the required firewall rules will come in one of the next minor releases.
+
+On the engine VM:
+```
+# stop the firewall
+systemctl stop firewalld
+systemctl disable firewalld
+# install ovirt-provider-ovn
+yum install -y ovirt-provider-ovn
+systemctl start ovirt-provider-ovn
+systemctl enable ovirt-provider-ovn
+```
+On the hosts:
+```
+# stop the firewall
+systemctl stop iptables
```

Comment:
Currently the engine and host-deploy are not able to manage OVN specific rules; we can to the user to add the required rules but they are going to be over-written by the engine if the user redeploys the host.
iptables handling for OVN will come with 4.1.1: https://gerrit.ovirt.org/#/c/72157